src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java

```diff
@@ -401,6 +407,7 @@ static String validateAndParseString(
     String output;
     if (input instanceof String) {
       output = (String) input;
+      verifyValidUtf8(output, columnName, "STRING");
```

Comment:
Any performance implication on this?

Reply:
Yeah, there is a performance impact. I ran some micro benchmarks on my machine and validation of 1GB of string data (1 million of 1KB strings) with Java 11 takes around 2.5 s. With Java 17 there is an improvement to 1.4 s.